src/main/org/codehaus/groovy/control/ResolveVisitor.java

```diff
@@ -852,6 +858,21 @@ protected Expression transformPropertyExpression(PropertyExpression pe) {
         return ret;
     }
 
+    private boolean isVisibleNestedClass(ClassNode type, ClassNode ceType) {
+        if (!type.isRedirectNode()) return false;
+        ClassNode redirect = type.redirect();
+        // public or private
+        if (Modifier.isPublic(redirect.getModifiers()) || Modifier.isProtected(redirect.getModifiers())) return true;
+        // package local
+        PackageNode classPackage = ceType.getPackage();
+        PackageNode nestedPackage = redirect.getPackage();
+        if ((redirect.getModifiers() & (Modifier.PROTECTED | Modifier.PUBLIC | Modifier.PRIVATE)) == 0 &&
+                ((classPackage == null && nestedPackage == null) ||
+                        classPackage != null && nestedPackage != null && classPackage.equals(nestedPackage)))
+            return true;
+        return false;
```

Comment:
I should probably use getName() on the package. It won't be private or the binary and will pick that up.